bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/api/AwattarApi.java

```diff
@@ -0,0 +1,155 @@
+
+    /**
+     * Exception thrown when the response from the aWATTar API is empty.
+     */
+    public class EmptyDataResponseException extends Exception {
+        private static final long serialVersionUID = 1L;
+
+        public EmptyDataResponseException(String message) {
+            super(message);
+        }
+    }
+
+    /**
+     * Constructor for the aWATTar API.
+     *
+     * @param httpClient the HTTP client to use
+     * @param zone the time zone to use
+     */
+    public AwattarApi(@NonNull HttpClient httpClient, @NonNull ZoneId zone) {
+        this.zone = zone;
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * Initialize the aWATTar API.
+     *
+     * @param config the configuration to use
+     */
+    public void initialize(AwattarBridgeConfiguration config) {
+        vatFactor = 1 + (config.vatPercent / 100);
+        basePrice = config.basePrice;
+
+        if (config.country.equals("DE")) {
+            this.url = URL_DE;
+        } else if (config.country.equals("AT")) {
+            this.url = URL_AT;
+        } else {
+            throw new IllegalArgumentException("Country code must be 'DE' or 'AT'");
+        }
+    }
+
+    /**
+     * Get the data from the aWATTar API.
+     * The data is returned as a sorted set of {@link AwattarPrice} objects.
+     * The data is requested from now minus one day to now plus three days.
+     *
+     * @return the data as a sorted set of {@link AwattarPrice} objects
+     * @throws InterruptedException if the thread is interrupted
+     * @throws TimeoutException if the request times out
+     * @throws ExecutionException if the request fails
+     * @throws EmptyDataResponseException if the response is empty
+     */
+    public SortedSet<AwattarPrice> getData()
+            throws InterruptedException, TimeoutException, ExecutionException, EmptyDataResponseException {
+        // we start one day in the past to cover ranges that already started yesterday
+        ZonedDateTime zdt = LocalDate.now(zone).atStartOfDay(zone).minusDays(1);
+        long start = zdt.toInstant().toEpochMilli();
+        // Starting from midnight yesterday we add three days so that the range covers
+        // the whole next day.
+        zdt = zdt.plusDays(3);
+        long end = zdt.toInstant().toEpochMilli();
+        StringBuilder request = new StringBuilder(url);
+        request.append("?start=").append(start).append("&end=").append(end);
+
+        logger.trace("aWATTar API request: = '{}'", request);
+        ContentResponse contentResponse = httpClient.newRequest(request.toString()).method(GET)
+                .timeout(10, TimeUnit.SECONDS).send();
+        int httpStatus = contentResponse.getStatus();
+        String content = contentResponse.getContentAsString();
+        logger.trace("aWATTar API response: status = {}, content = '{}'", httpStatus, content);
+
+        if (content == null) {
+            logger.error("The response from the aWATTar API is empty");
```

Comment:
This is not an `error`, at most a `warn` (error is reserved for core when the stability of the system is in danger). IMO the bridge should change it's state to `OFFLINE` if this happens, with a status detail "The response from the aWATTar API is empty". In that case no logging is needed, because the status change is logged.